### PR TITLE
Fix DynamicUser=yes with static User= whose UID and GID are different

### LINKED ISSUE #9702: systemd-resolved.service: Couldn't add UID/GID reference to unit, proceeding without: Device or resource busy
**systemd version the issue has been seen with**

239-7 in Debian unstable.

**Used distribution**

Debian unstable. I filed a bug report to the downstream as
[https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=904335](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=904335),
and I was suggested there reporting to the upstream.

**Expected behaviour you didn't see**

We don't see `systemd[1]: systemd-resolved.service: Couldn't add UID/GID reference to unit, proceeding without: Device or resource busy`

**Unexpected behaviour you saw**

The static user `systemd-resolve` in `/etc/passwd` and `DynamicUser=yes` give the following journal log.

```
Jul 23 12:02:57 debian-unstable systemd[1]: systemd-resolved.service: Couldn't add UID/GID reference to unit, proceeding without: Device or resource busy
```

Changing `DynamicUser=yes` to `DynamicUser=no` _fixes_ the problem.

To the down stream bug report, @mbiebl wrote:
>DynamicUser also takes care of setting up the runtime directory. Not
>sure if we can safely set that to false.
>In any case, systemd should gracefully deal with the fact that a system
>user has been allocated statically.

**Steps to reproduce the problem**
See above.

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #9702.

### CODE REVIEW

**Comment by yuwata** [comment]
Are there any static user on Ubuntu whose uid and gid are different? Currently, the new test is meaningless on Ubuntu.

**Comment by filbranden** [comment]
> Are there any static user on Ubuntu whose uid and gid are different? Currently, the new test is meaningless on Ubuntu.

Perhaps this one:

```
$ id games
uid=5(games) gid=60(games) groups=60(games)
```

Available on Ubuntu (18.04) and Debian (Stretch) it seems.

Also present on Fedora Rawhide, but it's a bit weirder there:

```
$ id games
uid=12(games) gid=100(users) groups=100(users)
$ getent group games
games:x:20:
```

😕 

In any case, seems to fit the bill.

I guess one option is to create one explicitly for your tests. I imagine the tmpfiles tests already create users, no? Might perhaps want to do the same here...

**Comment by yuwata** [comment]
Thanks. A test for Ubuntu or Debian is added which uses user games.

File: test/test-execute/exec-dynamicuser-fixeduser-adm.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Test DynamicUser with static User= whose uid and gid are different
+# On Fedora, user adm has uid==3 and gid==4.
+
+[Service]
+Type=oneshot
+ExecStart=/bin/sh -x -c 'test "$$(id -nG)" = "adm" && test "$$(id -ng)" = "adm" && test "$$(id -nu)" = "adm"'
```
Comment (keszybz) [inline test/test-execute/exec-dynamicuser-fixeduser-adm.service:7]: `sh` is enough ;)

**Comment by keszybz** [comment]
`boot-and-services    FAIL stderr: Failed to dump process list for 'systemd-nspawn@c1.service', ignoring: Unit systemd-nspawn@c1.service not loaded.`

**Comment by yuwata** [comment]
@poettering BTW, what do you think about also storing gid in the storage socket?

**Pull request merged**